dev-tools/test.sh

```diff
@@ -25,38 +26,51 @@ while test $# -gt 0; do
     esac
 done
 
-if [[ $CHANGED -gt 0 ]]; then
-    # if --changed flag was passed: only run changed tests and don't update dependency database
-    TESTMON="--testmon-nocollect"
-    CHANGED_MESSAGE=" affected by recent changes"
+# The default pytests args we use
+PYTEST_ARGS=("--quiet" "--disable-warnings" "--numprocesses=auto")
 
-    # check if .testmondata file exists; if not, inform that all tests will be run
-    if [ ! -f ".testmondata" ]; then
+# Check if --changed flag was passed
+if [[ -n "${CHANGED}" ]]; then
+    # Check if .testmondata file exists
+    if [[ -f ".testmondata" ]]; then
+        # Only run changed tests and don't update dependency database
+        PYTEST_ARGS+=("--testmon-nocollect")
+        CHANGED_MESSAGE=" affected by recent changes"
+    else
+        # Inform that all tests will be run
         echo -e "\nIt looks like you have not run pytest without the \"--changed\" flag before." | print_warning
         echo -e "Pytest has to build a dependency database by running all tests without the flag once.\n" | print_warning
+        # Override test path argument
         unset TEST_PATH
-        TESTMON="--testmon"
-        unset CHANGED_MESSAGE
+        # Tell testmon to run all tests and collect data
+        PYTEST_ARGS+=("--testmon-noselect")
     fi
 else
-    # else: run all tests, but update list of tests
-    TESTMON="--testmon-noselect"
+    # Run all tests, but update list of tests
+    PYTEST_ARGS+=("--testmon-noselect")
 fi
 
-if [[ -z "$TEST_PATH" ]]; then
-    echo -e "Running all tests$CHANGED_MESSAGE..." | print_info
-    deescalate_privileges pipenv run pytest $TESTMON --disable-warnings --quiet --numprocesses=auto --cov=integreat_cms --cov-report html
-elif [[ -e "$TEST_PATH" ]]; then
-    echo -e "Running tests in ${TEST_PATH}$CHANGED_MESSAGE..." | print_info
-    deescalate_privileges pipenv run pytest "$TEST_PATH" $TESTMON --quiet --disable-warnings --numprocesses=auto
-else
+# Determine whether coverage data should be collected
+if [[ -z "${CHANGED}" && -z "${TEST_PATH}" ]]; then
+    PYTEST_ARGS+=("--cov=integreat_cms" "--cov-report html")
```

Comment:
only collect coverage when neither the changed flag, nor the test path are given